api/src/controllers/assets.ts

```diff
@@ -152,11 +156,13 @@ router.get(
 
 		const vary = ['Origin', 'Cache-Control'];
 
-		const transformationParams: TransformationParams = res.locals['transformation'].key
+		const transformationPresets = res.locals['transformation'].key
 			? (res.locals['shortcuts'] as TransformationParams[]).find(
 					(transformation) => transformation['key'] === res.locals['transformation'].key
-			  )
-			: res.locals['transformation'];
+			  ) ?? {}
+			: {};
+
+		const transformationParams: TransformationParams = { ...transformationPresets, ...res.locals['transformation'] };
```

Comment:
We could make that more compact with just

```ts
		const transformationParams: TransformationParams = {
			...(res.locals['shortcuts'] as TransformationParams[]).find(
				(transformation) => transformation['key'] === res.locals['transformation'].key
			),
			...res.locals['transformation'],
		};
```

Reply:
`shurtcuts.find` can return undefined so trying to expand that with `...` directly may error.

Reply:
For Arrays yes but Objects do work with undefined

```js
{ ...undefined, ...{thisworks:123} } // {thisworks: 123}
```

Doesnt throw wheras

```js
[ ...undefined, ...[1,2] ] // Uncaught TypeError: undefined is not iterable
```

does throw

Reply:
Ah yeah right, `null` and `undefined` only fail on iterating not for object spreading 😂 what a language, they're both considered objects arent they.